src/App.tsx

```diff
@@ -10,38 +10,40 @@
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { actions as postsActions } from './features/posts/posts';
+import { actions as selectedPostActions } from './features/selectedPost/selectedPost'; // eslint-disable-line max-len
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
+  const author = useAppSelector(state => state.author);
+  const { items, loaded, hasError } = useAppSelector(state => state.posts);
+  const selectedPost = useAppSelector(state => state.selectedPost);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(postsActions.setLoader(false));
```

Comment:
The loader is set to `false` before fetching posts, which might be incorrect. Typically, you want to set the loader to `true` to indicate loading has started.